src/transformers/commands/pt_to_tf.py

```diff
@@ -225,8 +226,18 @@ def run(self):
 
         # Save the weights in a TF format (if needed) and confirms that the results are still good
         tf_weights_path = os.path.join(self._local_dir, TF_WEIGHTS_NAME)
+        pt_weights_path = os.path.join(self._local_dir, PT_WEIGHTS_NAME)
         if not os.path.exists(tf_weights_path) or self._new_weights:
             tf_from_pt_model.save_weights(tf_weights_path)
+            # If the resulting TF weights are much bigger than the PT weights (roughly 2x), then we are in the presence
+            # of float16 weights. We can set the corresponding backend, store the weights in the new format, and reset
+            # the backend to ensure the reloaded weights match their counterpart using 32 bit precision.
+            if float(os.path.getsize(tf_weights_path)) > 1.8 * os.path.getsize(pt_weights_path):
+                self._logger.warn("WEIGHT FORMAT UPDATE:float16 weights detected! Storing TF weights in float16")
```

Comment:
```suggestion
                self._logger.warn("WEIGHT FORMAT UPDATE: float16 weights detected! Storing TF weights in float16")
```

Reply:
> Will TensorFlow automatically load stored FP16 weights in FP32 like PyTorch does?

No, if we inspect the variables after loading, they are FP16. However, as soon as we pass some input, we can see that the outputs of the internal computations are FP32, despite the weights being FP16. 

The model error (vs PT) is exactly the same, before and after adding these lines :)

Reply:
The documentation is not clear, but I believe `tf.keras.backend.set_floatx` sets the precision of the internal computations. For instance, if we don't reset to float32 after storing as float16, the error is much much larger (~1e3 times larger).

Reply:
replied in the wrong place 🤦  moving the discusson